### ISSUE TITLE
Whitespace is clobbered

### ISSUE THREAD

**Issue by mqudsi** [neomake/neomake#1519]
When executing a non-customized maker script (something that just executes a command in the shell and returns plain-text output), it seems that at some point between command invocation and showing the results in the quickfix pane, whitespace output is clobbered.

Here's an example of what `cargo build` outputs to the terminal for a project:

```rust
error[E0244]: wrong number of type arguments: expected 0, found 1
  --> src/main.rs:41:41
   |
41 | fn not_found(_: &mut Request) -> Result<Request<R>, String> {
   |                                         ^^^^^^^^^^ expected no type arguments
```

And here's what neomake shows in the quickfix pane when executing the same:

```rust
src/main.rs|41 col 41 error 244| wrong number of type arguments: expected 0, found 1
|| |
|| 41 | fn not_found(_: &mut Request) -> Result<Request<R>, String> {
|| | ^^^^^^^^^^ expected no type arguments
```

What's funny/unexpected is that it's not leading whitespace that's being clobbered, but rather whitespace in the middle of the string, which isn't normally what gets lost.

**Comment by blueyed** [comment]
How do you run `cargo build` via Neomake?
Are you using the maker (i.e. `Neomake cargo`)?

**Comment by mqudsi** [comment]
No, I'm using the default maker (`Neomake!`)

```
Neomake: [18.18:1:1] Starting async job: ['/bin/bash', '-c', 'cargo build $*'].                                                                  Neomake: [18.18:1:2] Calling User autocmd NeomakeFinished with context: {'jobinfo': '…'}.                                                        Build complete  
```

**Comment by blueyed** [comment]
So via `&makeprg` then?
Please provide the whole log output - see the "new issue template".

**Comment by mqudsi** [comment]
Sure, here's the `:messages` for `Neomake!`:

```
"src/main.rs" 171L, 6555C                                                                                                              
The only match                                                                                                                         
The only match                                                                                                                         
Neomake [     ]: [1.-:1:1] Running makers: makeprg.                                                                                    
Neomake [     ]: [1.1:1:1] Calling User autocmd NeomakeJobInit with context: {'jobinfo': '…'}.                                         
Build started                                                                                                                          
Neomake [     ]: [1.1:1:1] Starting async job: ['/bin/bash', '-c', 'cargo build $*'].                                                  
Neomake [     ]: [1.1:1:1] cwd: /mnt/d/GIT/udpproxy.                                                                                   
Neomake [     ]: Skipping User autocmd NeomakeJobStarted: no hooks.                                                                    
Neomake [+0.26]: [1.1:1:1] stderr: makeprg: ['   Compiling libc v0.2.26', ''].                                                         
Neomake [     ]: [1.1:1:1] stderr: makeprg: ['   Compiling getopts v0.2.14', ''].                                                      
Neomake [+1.26]: [1.1:1:1] stderr: makeprg: ['   Compiling rand'].                                                                     
Neomake [     ]: [1.1:1:1] stderr: makeprg: [' v0.3.15', ''].                                                                          
Neomake [+1.83]: [1.1:1:1] stderr: makeprg: ['   Compiling '].                                                                         
Neomake [     ]: [1.1:1:1] stderr: makeprg: ['udpproxy v0.1.0 (file:///mnt/d/GIT/udpproxy)', ''].                                      
Neomake [+0.13]: [1.1:1:1] stderr: makeprg: ['error: expected identifier, found keyword `use`', '  --> src/main.rs:15:1', '   |', '15 | use std::env;', '   | ^^^', '', 'error: aborting due to previous error(s)', '', ''].                                                  
Neomake [     ]: [1.1:1:1] stderr: makeprg: ['error: '].                                                                               
Neomake [     ]: [1.1:1:1] stderr: makeprg: ['Could not compile `udpproxy`.', '', 'To learn more, run the command again with --verbose.', ''].                                                                                                                                
Neomake [     ]: [1.1:1:1] exit: makeprg: 101.                                                                                         
Neomake [     ]: [1.1:1:1] makeprg: processing 15 lines of output.                                                                     
Neomake [     ]: [1.1:1:1] Creating quickfix list.                                                                                     
Neomake [     ]: Modified list entry (postprocess): text: '   |' => '|'.                                                               
Neomake [     ]: Modified list entry (postprocess): text: '   | ^^^' => '| ^^^'.                                                       
Neomake [     ]: [1.1:1:1] Processing 4 entries.                                                                                       
Neomake [+0.01]: Placing sign: sign place 5000 line=15 name=neomake_project_err buffer=1.                                              
Neomake [     ]: [1.1:1:1] Skipped 3 entries without bufnr.                                                                            
Neomake [     ]: Skipping User autocmd NeomakeCountsChanged: no hooks.                                                                 
Neomake [     ]: Handling quickfix list: executing cwindow.                                                                            
Neomake [+0.07]: [1.1:1:2] makeprg: completed with exit code 101.                                                                      
Neomake [     ]: [1.1:1:2] Cleaning jobinfo.                                                                                           
Neomake [     ]: Skipping User autocmd NeomakeJobFinished: no hooks.                                                                   
Neomake [     ]: Removing signs: {file: {}, project: {}}.                                                                              
Neomake [     ]: Handling quickfix list: executing cwindow.                                                                            
Neomake [     ]: [1.1:1:2] Calling User autocmd NeomakeFinished with context: {'jobinfo': '…'}.                                        
Build complete                                                                                                                         
Neomake [     ]: [1.-:-:2] Cleaning make info.                                                                                         
```

and `NeomakeInfo`:

#### Neomake debug information
Async support: 1
Current filetype: rust
Windows: 0
[shell, shellcmdflag, shellslash]: ['/bin/bash', '-c', 0]

##### Enabled makers
For the current filetype ("rust", used with :Neomake):
 - cargo
   - append_file: 0
   - args: ['check', '--message-format=json', '--quiet']
   - auto_enabled: 1
   - cwd: '%:p:h'
   - errorformat: '[%t%n] "%f" %l:%v %m,[%t] "%f" %l:%v %m'
   - exe: 'cargo'
   - process_output: function('neomake#makers#ft#rust#CargoProcessOutput')
NOTE: you can define g:neomake_rust_enabled_makers to configure it (or b:neomake_rust_enabled_makers).

For the project (used with :Neomake!):
 - makeprg
   - args: []
   - auto_enabled: 0
   - errorformat: '%f:%l:%c: %t%*[^:]: %m,%f:%l:%c: %*\d:%*\d %t%*[^:]: %m,%-G%f:%l %s,%-G%*[ ]^,%-G%*[ ]^%*[~],%-G%*[ ]...,%-G,%-Gerro
r: aborting %.%#,%-Gerror: Could not compile %.%#,%Eerror: %m,%Eerror[E%n]: %m,%Wwarning: %m,%Inote: %m,%C %#--> %f:%l:%c,%-G%\s%#Downl
oading%.%#,%-G%\s%#Compiling%.%#,%-G%\s%#Finished%.%#,%-G%\s%#error: Could not compile %.%#,%-G%\s%#To learn more\,%.%#'
   - exe: 'makeprg'
   - fn: function('7')
   - ERRORS:
     - maker's exe (makeprg) is not executable.
NOTE: you can define g:neomake_enabled_makers to configure it.

Default maker settings:
 - remove_invalid_entries: 0
 - buffer_output: 1

##### Settings                                                                                                                         ```                                                                                                                                    g:neomake_cpp_clangtidy_maker = {'errorformat': '%E%f:%l:%c: fatal error: %m,%E%f:%l:%c: error: %m,%W%f:%l:%c: warning: %m,%-G%\m%\%%(LLVM ERROR:%\|No compilation database found%\)%\@!%.%#,%E%m', 'args': ['--', '-fsyntax-only', '-Wall', '-Wextra', '-std=c++1z', '-I./'], 'exe': 'clang-tidy'}                                                                                                                  g:neomake_cpp_clangxx_maker = {'errorformat': '%-G%f:%s:,%f:%l:%c: %trror: %m,%f:%l:%c: %tarning: %m,%I%f:%l:%c: note: %m,%f:%l:%c: %m,%f:%l: %trror: %m,%f:%l: %tarning: %m,%I%f:%l: note: %m,%f:%l: %m', 'args': ['-fsyntax-only', '-Wall', '-Wextra', '-std=c++1z', '-I./'], 'exe': 'clang++'}                                                                                                                    g:neomake_cpp_enabled_makers = ['clangxx', 'clangtidy']                                                                                g:neomake_enabled_makers = ['makeprg']                                                                                                 g:neomake_makeprg_maker = {'errorformat': '%*[^"]"%f"%*\D%l: %m,"%f"%*\D%l: %m,%-G%f:%l: (Each undeclared identifier is reported only once,%-G%f:%l: for each function it appears in.),%-GIn file included from %f:%l:%c:,%-GIn file included from %f:%l:%c\,,%-GIn file included from %f:%l:%c,%-GIn file included from %f:%l,%-G%*[ ]from %f:%l:%c,%-G%*[ ]from %f:%l:,%-G%*[ ]from %f:%l\,,%-G%*[ ]from %f:%l,%f:%l:%c:%m,%f(%l):%m,%f:%l:%m,"%f"\, line %l%*\D%c%*[^ ] %m,%D%*\a[%*\d]: Entering directory %*[`'']%f'',%X%*\a[%*\d]: Leaving directory %*[`'']%f'',%D%*\a: Entering directory %*[`'']%f'',%X%*\a: Leaving directory %*[`'']%f'',%DMaking %*\a in %f,%f|%l| %m', 'args': ['-j4'], 'exe': 'make'}                                                                                                                       g:neomake_open_list = 1                                                                                                                g:neomake_place_signs = 1                                                                                                              g:neomake_verbose = 3                                                                                                                  ```                                                                                                                                                                                                                                                                           #### :version                                                                                                                          ```                                                                                                                                    NVIM v0.2.1-696-gbdc72c7                                                                                                               Build type: RelWithDebInfo                                                                                                             Compilation: /usr/bin/cc -Wconversion -DNVIM_MSGPACK_HAS_FLOAT32 -O2 -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -I/home/mqudsi/rand/neovim/build/config -I/home/mqudsi/rand/neovim/src -I/home/mqudsi/rand/neovim/.deps/usr/include -I/home/mqudsi/rand/neovim/.deps/usr/include -I/home/mqudsi/rand/neovim/.deps/usr/include -I/home/mqudsi/rand/neovim/.deps/usr/include -I/home/mqudsi/rand/neovim/.deps/usr/include -I/home/mqudsi/rand/neovim/.deps/usr/include -I/usr/include -I/home/mqudsi/rand/neovim/build/src/nvim/auto -I/home/mqudsi/rand/neovim/build/include                                                                                                          Compiled by mqudsi@Blitzkrieg                                                                                                                                                                                                                                                 Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui                                                             For differences from Vim, see :help vim-differences                                                                                                                                                                                                                              system vimrc file: "$VIM/sysinit.vim"                                                                                                 fall-back for $VIM: "/usr/local/share/nvim"                                                                                          ```                                                        

#### :messages                                                                                                                         ```                                                                                                                                    E325: ATTENTION                                                                                                                        "src/main.rs" 171L, 6555C                                                                                                              NVIM v0.2.1-696-gbdc72c7                                                                                                               Build type: RelWithDebInfo                                                                                                             Compilation: /usr/bin/cc -Wconversion -DNVIM_MSGPACK_HAS_FLOAT32 -O2 -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -I/home/mqudsi/rand/neovim/build/config -I/home/mqudsi/rand/neovim/src -I/home/mqudsi/rand/neovim/.deps/usr/include -I/home/mqudsi/rand/neovim/.deps/usr/include -I/home/mqudsi/rand/neovim/.deps/usr/include -I/home/mqudsi/rand/neovim/.deps/usr/include -I/home/mqudsi/rand/neovim/.deps/usr/include -I/home/mqudsi/rand/neovim/.deps/usr/include -I/usr/include -I/home/mqudsi/rand/neovim/build/src/nvim/auto -I/home/mqudsi/rand/neovim/build/include                                                                                                          ```

**Comment by mqudsi** [comment]
*please note that I updated my comment above after removing my custom maker to help make debugging easier*

**Comment by blueyed** [comment]
Fixed by https://github.com/neomake/neomake/commit/4105d31f7363c1ae0e87ac1d2ae240803c2fa524 I guess.
